src/libDirectoryService/FinalBlockPostProcessing.cpp

```diff
@@ -535,6 +536,8 @@ bool DirectoryService::ProcessFinalBlockConsensus(
         LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
                   "Process final block arrived early, saved to buffer");
 
+        lock_guard<mutex> g(m_mutexConsensus);
```

Comment:
Not sure whether is it necessary here. Any insight on this?

Reply:
For every invoking of ProcessFinalBlockConsensus, when it comes to ProcessMessage and finally enters the WhenDone function, it will be covered by this mutex, it protect the race condition of the changes on m_consensusID and so on. While the following code deals with the m_consensusID, and eventually tries to RunConsensus, it's necessary to have this mutex for protection